src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -0,0 +1,474 @@
+            f"Duplicated columns:{str(get_duplicated_columns(dataset))}"
+        )
+        return {"result": False, "msg": msg}
+    else:
+        msg = f"{namespace} dataset has been validated."
+        return {"result": True, "msg": msg}
+
+
+def compare_datasets_by_data_type(
+    parquet_bucket: str,
+    staging_namespace: str,
+    main_namespace: str,
+    s3_filesystem: fs.S3FileSystem,
+    data_type: str,
+) -> str:
+    """This runs the bulk of the comparison functions from beginning to end by data type
+
+    Args:
+        parquet_bucket (str): name of the bucket containing the parquet datasets
+        staging_namespace (str): name of namespace containing the "new" data
+        main_namespace (str): name of namespace containing the "established" data
+        s3_filesystem (fs.S3FileSystem): filesystem instantiated by aws credentials
+        data_type (str): data type to be compared for the given datasets
+
+    Returns:
+        str: final report on the datasets for the given data type
+    """
+    header_msg = (
+        f"\n\nParquet Dataset Comparison running for Data Type: {data_type}"
+        f"\n-----------------------------------------------------------------\n\n"
+    )
+    staging_dataset = get_parquet_dataset(
+        dataset_key=f"s3://{parquet_bucket}/{staging_namespace}/parquet/{data_type}/",
+        s3_filesystem=s3_filesystem,
+    )
+    main_dataset = get_parquet_dataset(
+        dataset_key=f"s3://{parquet_bucket}/{main_namespace}/parquet/{data_type}/",
+        s3_filesystem=s3_filesystem,
+    )
+    # go through specific validation for each dataset prior to comparison
+    staging_is_valid_result = is_valid_dataset(staging_dataset, staging_namespace)
+    main_is_valid_result = is_valid_dataset(main_dataset, main_namespace)
+    if (
+        staging_is_valid_result["result"] == False
+        or main_is_valid_result["result"] == False
+    ):
+        comparison_report = (
+            f"{staging_is_valid_result['msg']}\n{main_is_valid_result['msg']}"
+        )
+    # check that they have columns in common to compare
+    elif not has_common_cols(staging_dataset, main_dataset):
+        comparison_report = (
+            f"{staging_namespace} dataset and {main_namespace} have no columns in common."
+            f" Comparison cannot continue."
+        )
+    else:
+        comparison_report = compare_datasets_and_output_report(
+            data_type=data_type,
+            staging_dataset=staging_dataset,
+            main_dataset=main_dataset,
+            staging_namespace=staging_namespace,
+            main_namespace=main_namespace,
+        )
+
+        comparison_report = add_additional_msg_to_comparison_report(
+            comparison_report,
+            add_msgs=compare_column_names(data_type, staging_dataset, main_dataset),
+            msg_type="column_name_diff",
+        )
+    return f"{header_msg}{comparison_report}"
+
+
+def main():
```

Comment:
At this point, I'm thinking this main function's logic (just this one if-else statement) would just have to be tested with some test data rather than writing any test functions just because it would be hard to even compare the results since it's a large chunk of string. Every function here in `main` is tested in their own tests.

Reply:
I agree with your assessment.  An integration test is most likely easiest here instead of trying to mock. (1st option)